### PR TITLE
Feature/kyv 791 kassalla kayttaja siirretaan valittuun paytrail maksutapaan automaattisesti maksun luonnin yhteydessa

### PR DESCRIPTION
**Pull request by severikupari1**

### CODE REVIEW

File: src/talons/checkout/usePaymentMethods.ts [hunk 1]
```diff
@@ -85,15 +86,42 @@ export const usePaymentMethods = () => {
...
     function proceedToPayment() {
-      const { payment: { paymentUrl } } = paymentRequestData;
+      const { payment: { paymentUrl, paymentGateway, paytrailProvider }} = paymentRequestData;
       setProceedToPaymentCalled(true);
       // Sets loading spinner to show before reloading
       setLoading(true)
-      // Using location.href because it faster than assign
-      window.location.href = paymentUrl;
+
+      if (paymentGateway === PaymentGateway.PAYTRAIL.toString()) {
```
Comment (martinldev) [inline src/talons/checkout/usePaymentMethods.ts:94]: I dont think it is necessary to call toString() here. It should return the mapped string value with only "PaymentGateway.PAYTRAIL".

File: src/talons/checkout/usePaymentMethods.ts [hunk 2]
```diff
@@ -85,15 +86,42 @@ export const usePaymentMethods = () => {
...
+            paytrailForm.appendChild(inputElement)
+          }
+        );
+
+        document.body.appendChild(paytrailForm);
+        paytrailForm.submit();
+      }
+
+      // Added fallback to visma if empty payment gateway
+      if (paymentGateway === PaymentGateway.VISMA.toString() || !paymentGateway) {
```
Comment (martinldev) [inline src/talons/checkout/usePaymentMethods.ts:117]: Same here as above.

**Review by martinldev — APPROVED** [review]
Some very small things in comments. Other than that it looks very good!

File: src/talons/checkout/usePaymentMethods.ts (hunk 1 above)
Comment (severikupari1) [reply src/talons/checkout/usePaymentMethods.ts:94]: It is needed to call to string, it does not work as same in java

**Pull request merged**